plugins/fs/fs.c

```diff
@@ -106,13 +122,26 @@ run(LuastatusPluginData *pd, LuastatusPluginRunFuncs funcs)
     while (1) {
         // make a call
         lua_State *L = funcs.call_begin(pd->userdata);
-        lua_createtable(L, 0, p->paths.size);
+        lua_newtable(L);
         for (size_t i = 0; i < p->paths.size; ++i) {
             const char *path = p->paths.data[i];
             if (push_for(pd, L, path)) {
                 lua_setfield(L, -2, path);
             }
         }
+        for (size_t i = 0; i < p->globs.size; ++i) {
+            glob_t gbuf;
+            int err = glob(p->globs.data[i], GLOB_NOSORT, NULL, &gbuf);
+            if (err == GLOB_NOSPACE) {
```

Comment:
Rather,
```c
switch (glob(p->globs.data[i], GLOB_NOSORT, NULL, &gbuf)) {
case 0:
case GLOB_NOMATCH:
    break;
default:
    LS_WARNF(pd, "glob() failed (out of memory?)");
}
```

Remember systems may invent their own reasons for `glob()` to fail (although this is not too probable to do more sophiscated error reporting).